dds/DCPS/XTypes/TypeLookupService.h

```diff
@@ -40,6 +44,9 @@ class OpenDDS_Dcps_Export TypeLookupService : public virtual DCPS::RcObject {
   void update_type_identifier_map(const TypeIdentifierPairSeq& tid_pairs);
   bool complete_to_minimal_type_object(const TypeObject& cto, TypeObject& mto) const;
 
+  typedef OPENDDS_MAP(TypeIdentifier, DynamicType_rch) DynamicTypeMap;
+  DynamicType_rch complete_to_dynamic(const CompleteTypeObject&, DynamicTypeMap& dt_map) const;
```

Comment:
Can we remove the parameter `dt_map` in `complete_to_dynamic()` and instead have a private member of type `DynamicTypeMap` for caching the computed mappings instead? That way callers won't have to worry about passing a `dt_map` argument, and it can be completely internal to the implementation.

Another advantage is that multiple calls to `complete_to_dynamic()` may share some common mappings, and if so a later call doesn't have to recompute a common mapping again.

Reply:
Recent changes made it so I had to move the creation of DynamicTypeMap to outside where I call into complete_to_dynamic so that the weak_rch don't destroy themselves early. It was a compromise that had to be made so we don't have memory leaks, but also don't have failures due to early deletion where nothing held the rch.